src/rt/aaA.d

```diff
@@ -684,6 +684,10 @@ extern (C) pure nothrow @nogc
                 break;
         }
     }
+    size_t _aaRangeLen(Range r)
+    {
+        return r.impl is null ? 0 : r.impl.length;
```

Comment:
This is wrong. The length should reflect how many items are left in the range, not the total elements in the AA.